From b9b3ebeba33cd6ba59145368550cf6ff771ac6d4 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Tue, 26 Sep 2017 10:57:34 +0200
Subject: [PATCH 3/4] qemu-char: add cyrillic characters 'numerosign' to VNC
 keysyms

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <20170926105734.746-4-kraxel@redhat.com>
Patchwork-id: 76564
O-Subject: [RHEL-7.5 qemu-kvm PATCH 3/3] qemu-char: add cyrillic characters 'numerosign' to VNC keysyms
Bugzilla: 1476641
RH-Acked-by: Laurent Vivier <lvivier@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: John Snow <jsnow@redhat.com>

From: Wang Xin <wangxinxin.wang@huawei.com>

This patch adds missing cyrillic character 'numerosign' to the VNC
keysym table, it's needed by Russian keyboard. And I get the keysym from
'<X11/keysymdef.h>', the current keysym table in Qemu was generated from
it.

Signed-off-by: Wang xin <wangxinxin.wang@huawei.com>
Signed-off-by: Gonglei <arei.gonglei@huawei.com>
Signed-off-by: Michael Tokarev <mjt@tls.msk.ru>
(cherry picked from commit 09f4fbe47314c2f13f16bf081997611f7f4251ea)
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 ui/vnc_keysym.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ui/vnc_keysym.h b/ui/vnc_keysym.h
index 1dc039f..7fa2bc1 100644
--- a/ui/vnc_keysym.h
+++ b/ui/vnc_keysym.h
@@ -404,6 +404,7 @@ static const name2keysym_t name2keysym[]={
 {"breve",                         0x01a2},  /* U+02D8 BREVE */
 {"caron",                         0x01b7},  /* U+02C7 CARON */
 {"Ccaron",                        0x01c8},  /* U+010C LATIN CAPITAL LETTER C WITH CARON */
+{"numerosign",                    0x06b0},  /* U+2116 NUMERO SIGN */
 {"Cyrillic_a",                    0x06c1},  /* U+0430 CYRILLIC SMALL LETTER A */
 {"Cyrillic_A",                    0x06e1},  /* U+0410 CYRILLIC CAPITAL LETTER A */
 {"Cyrillic_be",                   0x06c2},  /* U+0431 CYRILLIC SMALL LETTER BE */
-- 
1.8.3.1

